### PR TITLE
Allow passing parameters of non-model types into generated client methods

### PR DESCRIPTION
**Pull request by oleksandrpravosudko-okta**
* derive weak types from models to be used for passing into Client methods
* remove IRequestExecutor interface
* add `url` property to  RequestOptions interface

### CODE REVIEW

File: src/types/client.d.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import { GeneratedApiClient } from './generated-client';
 import { OAuth } from './oauth';
 import { Http } from './http';
-import { IRequestExecutor } from './request-executor';
+import { RequestExecutor } from './request-executor';
```
Comment (shuowu) [inline src/types/client.d.ts:16]: Is this a type name change? Looks like a breaking change

File: README.md [hunk 2]
```diff
@@ -769,6 +769,30 @@ const client = new okta.Client({
 })
 ```
 
+### TypeScript usage examples (>=4.5.0)
```
Comment (shuowu) [inline README.md:772]: Can you add one more example for the newly introduced `optional` model type?

File: src/types/models/AcsEndpoint.d.ts [hunk 3]
```diff
@@ -25,6 +26,9 @@ declare class AcsEndpoint extends Resource {
 
 }
 
+type AcsEndpointOptionsType = OptionalKnownProperties<AcsEndpoint>;
+
 export {
-  AcsEndpoint
+  AcsEndpoint,
+  AcsEndpointOptionsType
```
Comment (shuowu) [inline src/types/models/AcsEndpoint.d.ts:33]: How about name it as `xxxOptionalType`?

**Comment by shuowu** [comment]
Looks good to me in general. Can we put the type breaking change (`remove IRequestExecutor interface`) for next major version update?

File: test/type/client.test-d.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { expectError, expectType } from 'tsd';
 import { Response } from 'node-fetch';
 import { Client } from '../../src/types/client';
 import { Collection } from '../../src/types/collection';
-import { Application } from '../../src/types/models/Application';
+import { Application, ApplicationOptionsType } from '../../src/types/models/Application';
```
Comment (shuowu-okta) [inline test/type/client.test-d.ts:5]: should it be ApplicationOptions?

File: test/type/client.test-d.ts [hunk 5]
```diff
@@ -16,3 +16,9 @@ expectError(client.deleteApplicationUser('appId', 'userId', {sendEmail: 0}));
 // Client methods return either Promise or Collection
 expectType<Promise<Response>>(client.deletePolicy('policyId'));
 expectType<Collection<Application>>(client.listApplications());
+
+// methods expecting body request parameters
+const appOptions: ApplicationOptionsType = {
```
Comment (shuowu-okta) [inline test/type/client.test-d.ts:21]: same here

File: README.md [hunk 6]
```diff
@@ -769,6 +769,57 @@ const client = new okta.Client({
...
+  "signOnMode": "BOOKMARK",
+  "settings": {
+    "app": {
+      "requestIntegration": false,
+      "url": "https://example.com/bookmark.htm"
+    }
+  }
+};
+
+client.createApplication(bookmarkApp).then((createdApp: Application) => {
```
Comment (shuowu-okta) [inline README.md:818]: nit: bookmarkApp -> bookmarkAppOptions

**Review by shuowu-okta — APPROVED** [review]

**Pull request closed without merging**